### ISSUE TITLE
Question: Using Connect-MgGraph to authenticate as user (delegated permission) in script, without interraction (needed in script).

### ISSUE THREAD

**Issue by zaicnupagadi** [microsoftgraph/msgraph-sdk-powershell#1168]
Hi,

I got issue a little bit similar to https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/691, however instead of an app I need to use a user credentials. Let me explain why: I would like to use MS Bookings PowerShell to add staff member (New-MgBookingBusinessStaffMember), however to perfom this command, a user that runs this need to have full access on the mailbox - this is why I cannot use app registration. The only way I've found so far is device token - but this is not the solution as this has some certain TTL/cache on the machine might be cleared/machine restarted etc.

TL;DR:
I would like to use command Connect-MgGraph with "-Credential" parameter, or something else that would enter me Graph as specific user (as that user has FullAccess permissions to the Booking mailbox) - is it possible?

Apprecite any help on this,
Pawel Jarosz

**Comment by zaicnupagadi** [comment]
I have found it! http://laurierhodes.info/?q=node/137 So here I generate a token as a given user leveraging particular app registrar, and later that token can be used to log in, in Connect-MgGraph. Exactly what I needed.
